### DISCUSSION TITLE
Mobile Application running Oqtane

### DISCUSSION THREAD

**Discussion by jsboige** [oqtane/oqtane.framework#1118] [Ideas] [answered]
Has there been any attempt to run Oqtane as a native mobile application ?
There seem to be those 2 pathways:
- [Official mobile blazor bindings](https://docs.microsoft.com/fr-fr/mobile-blazor-bindings/) ([repo](https://github.com/dotnet/MobileBlazorBindings))
- [BlazorMobile](https://github.com/Daddoon/BlazorMobile)
As far as I can tell, running the client with proper external web API configuration should be possible, maybe at not so much of a cost. With modular native apps composed through web services, that would possibly significantly increase Oqtane's appeal for a large audience.

**Comment by sbwalker** [comment, marked as answer]
Oqtane currently supports both official Blazor hosting models - Server and WebAssembly. It can also be configured to run as Progessive Web Application (PWA). Based on its client/server architecture it should also be possible to run it as a native mobile app or desktop application - it would just need the appropriate hosting container. At this point the Blazor Mobile bindings are still “experimental” and BlazorMobile does not appear to support .NET 5. I have not experimented with running Oqtane on either option - but I agree that it would be a great feature.

**Reply by jsboige** [reply]
Thanks, I look forward to that being an option, and I'll be happy to help.
